common/unordered_inline_set.h

```diff
@@ -0,0 +1,741 @@
+        auto it = find(x);
+        if (it == end()) return {it, it};
+        return {it, it.get_next()};
+    }
+    template< class K >
+    std::pair<const_iterator, const_iterator> equal_range( const K& x ) const {
+        auto it = find(x);
+        if (it == end()) return {it, it};
+        return {it, it.get_next()};
+    }
+
+public:
+    iterator find( const Key& key ) {
+        auto r = _find(key);
+        return r.second ? end() : r.first;
+    }
+    const_iterator find( const Key& key ) const {
+        auto r = _find(key);
+        return r.second ? end() : r.first;
+    }
+    template< class K >
+    iterator find( const K& x ) {
+        auto r = _find(x);
+        return r.second ? end() : r.first;
+    }
+    template< class K >
+    const_iterator find( const K& x ) const {
+        auto r = _find(x);
+        return r.second ? end() : r.first;
+    }
+
+    iterator erase( iterator pos ) {
+        if (pos._a != &_data ||
+            pos._i >= _data.capacity())
+                return pos;
+        (pos++).erase();
+        _check_down_rehash(pos);
+        return pos;
+    }
+    // iterator erase( const_iterator pos );
+    iterator erase( const_iterator first, const_iterator last ) {
+        assert(first._a == _data);
+        assert( last._a == _data);
+        while (first != last)
+            (first++).erase();
+        return first;
+    }
+protected:
+    void _rehash( size_type count, iterator& pos ) {
+        assert(pos._a == &_data && pos._i <= _data.capacity());
+        auto at_least = std::ceil(_data.size() / max_load_factor());
+        if (count < at_least) count = at_least;
+        if (size() == count) return;
+        auto data = std::move(_data);
+        _data.reset(count);
+        _set_load_min_max();
+        for (auto it = data.begin(); it != data.end(); ++it)
+            insert(it.extract());
+    }
+    void _check_down_rehash(iterator pos) {
+        if (_data.size() >= _load_min ||
+            _data.capacity() <= MIN_CAPACITY) return;
+
+        rehash(_data.capacity() / 2);
+        _load_min /= 2;
+        _load_max /= 2;
+
+    }
+    void _check_down_rehash() {
+        if (_data.size() <= _load_max ) return;
+        rehash(_data.capacity() * 2);
+        _load_min *= 2;
+        _load_max *= 2;
+    }
+    size_type _erase_it(iterator it) {
+        if (it != end()) {
+            erase(it);
+            return 1;
+        } else {
+            return 0;
+        }
+    }
+public:
+    size_type erase( const Key& key ) {
+        return _erase_it(find(key));
+    }
+    template< class K >
+    size_type erase( K&& k ) {
+        return _erase_it(find(std::forward<K>(k)));
+    }
+
+    void swap( unordered_inline_set& other ) noexcept;
+
+    template< class K >
+    value_type&& extract( K&& x ) {
+        return extract(find(std::forward<K>(x)));
+    }
+    value_type&& extract( iterator position ) {
+        assert(position._a == &_data);
```

Comment:
`position._a` `position._i` are protected field of `iterator` type
and `get_bit` is protected